nucleof072rb/Core/Src/main.c

```diff
@@ -1,181 +1,217 @@
+  *
+  * This software component is licensed by ST under BSD 3-Clause license,
+  * the "License"; You may not use this file except in compliance with the
+  * License. You may obtain a copy of the License at:
+  *                        opensource.org/licenses/BSD-3-Clause
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+#include "spi.h"
+#include "tim.h"
+#include "usart.h"
+#include "gpio.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  /* USER CODE BEGIN 2 */
+  const uint16_t MAX_ADC_VALUE = 1023; // max num from 10bits
+  // uint16_t COUNTER_PERIOD = 64000;
+  const uint16_t MAX_CYCLE_COUNTS = 6400; // 10% of counter period
+  const uint16_t MIN_CYCLE_COUNTS = 3200; // 5% of counter period
+
+  uint8_t transmission_data_buffer[3] = {0b00000001, 0b10000000, 0};
+  uint8_t reception_data_buffer[3] = {0, 0, 0};
+
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+
+	  // Pull down CS line to start SPI communication
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  // Get the data
+	  HAL_SPI_TransmitReceive(&hspi1, transmission_data_buffer, reception_data_buffer, 3, 0);

```

Comment:
3 and 0 are magic numbers! Please make them consts or #define

Also, it looks like you are setting a timeout of 0ms for this transfer and receive function. This is telling the function that you want it to wait for 0ms to transfer and receive the data over spi before quitting... Basically, it will do nothing.

Be careful on these things, the only reason I caught this is because I am very familiar with this function but many times people who are not familiar with the code will review your code, so it is up to you to do the dudiligence and make sure you understand every parameter of a function you use. If you don't know, just ask :)